src/firewall/core/fw_config.py

```diff
@@ -174,6 +174,41 @@ def update_firewalld_conf(self):
         else:
             self._firewalld_conf.read()
 
+    # helper funtion (for detecting if builtin outside this module without throwing an error)
+
+    def is_builtin(self, obj):
+        if obj.builtin or not obj.default:
+            return True
+        return False
+
+    # Reset defaults (for offline mode)
+
+    def reset_defaults(self):
```

Comment:
as mentioned, should I introduce a parameter to check if the method is running in offline mode? The only valid places would be here or the cli to get access to FirewallD's reload() functionality, unless resetDefaults() (client.py) was moved to FirewallClient